pkg/util/storage_client.go

```diff
@@ -0,0 +1,46 @@
+package util
+
+import (
+	"time"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Constants for storage client configuration
+const (
+	DefaultRegion         = "us-east-1"
+	DefaultRequestTimeout = 15 * time.Second
+)
+
+// StorageClientParameters holds configuration for S3/IAM clients.
+type StorageClientParameters struct {
+	AccessKeyID     string
+	SecretAccessKey string
+	Endpoint        string
+	Region          string
+	TLSCert         []byte // Optional field for TLS certificates
+	Debug           bool   // Optional field for debug mode
+}
+
+// NewStorageClientParameters initializes default storage client parameters.
+func NewStorageClientParameters() StorageClientParameters {
```

Comment:
nitpicking: technically a `NewXX` function should return a pointer to an object per OOP habits, to avoid surprising the caller.